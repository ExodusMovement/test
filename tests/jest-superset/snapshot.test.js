// Main test file is tests/jest/snapshot.test.js
// This one just additionally rechecks named snapshot presence

it('supports named snapshots', async () => {
  expect({ name: 'Bruce Wayne' }).toMatchSnapshot('public knowledge')
  expect({ alterEgo: 'Batman', name: 'Bruce Wayne' }).toMatchSnapshot('not so public knowledge')
  expect({ name: 'Joker', address: 'Arkham Asylum' }).toMatchSnapshot('public knowledge')

  let createRequire
  try {
    ;({ createRequire } = await import('node:module'))
  } catch {
    // skip the rest of this test for environments without node:module
    return
  }

  const require = createRequire(import.meta.url)
  const snapshots = require('./__snapshots__/snapshot.test.js.snap')

  ;[
    'supports named snapshots: public knowledge 1',
    'supports named snapshots: public knowledge 2',
    'supports named snapshots: not so public knowledge 1',
  ].forEach((name) => {
    expect(Object.hasOwn(snapshots, name)).toBe(true)
  })
})
