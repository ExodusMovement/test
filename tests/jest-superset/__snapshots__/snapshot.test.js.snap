exports[`supports named snapshots: not so public knowledge 1`] = `
{
  "alterEgo": "Batman",
  "name": "Bruce Wayne",
}
`;

exports[`supports named snapshots: public knowledge 1`] = `
{
  "name": "Bruce Wayne",
}
`;

exports[`supports named snapshots: public knowledge 2`] = `
{
  "address": "Arkham Asylum",
  "name": "Joker",
}
`;
